drivers/accel/adxl355/iio_adxl355.c

```diff
@@ -0,0 +1,1117 @@
+ * @param reg	   - Address of the register to be written to.
+ * @param writeval - Data to be written.
+ *
+ * @return ret    - Result of the writing procedure.
+*******************************************************************************/
+static int adxl355_iio_write_reg(struct adxl355_iio_dev *dev, uint32_t reg,
+				 uint32_t writeval)
+{
+	uint8_t val = writeval;
+	return adxl355_write_device_data(dev->adxl355_dev, reg, 1, &val);
+}
+
+/***************************************************************************//**
+ * @brief Handles the read request for raw attribute.
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl355_iio_read_raw(void *dev, char *buf, uint32_t len,
+				const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t ret;
+	uint16_t temp_raw;
+	uint32_t accel_x;
+	uint32_t accel_y;
+	uint32_t accel_z;
+	struct adxl355_iio_dev *iio_adxl355;
+	struct adxl355_dev *adxl355;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl355 = (struct adxl355_iio_dev *)dev;
+
+	if (!iio_adxl355->adxl355_dev)
+		return -EINVAL;
+
+	adxl355 = iio_adxl355->adxl355_dev;
+
+	switch (channel->type) {
+	case IIO_ACCEL:
+		ret = adxl355_get_raw_xyz(adxl355, &accel_x, &accel_y, &accel_z);
+		if (ret)
+			return ret;
+
+		switch (channel->address) {
+		case ACCEL_AXIS_X:
+			ret = sign_extend32(accel_x, 19);
+			break;
+		case ACCEL_AXIS_Y:
+			ret = sign_extend32(accel_y, 19);
+			break;
+		case ACCEL_AXIS_Z:
+			ret = sign_extend32(accel_z, 19);
+			break;
+
+		default:
+			return -EINVAL;
+		}
+
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &ret);
+
+	case IIO_TEMP:
+		ret = adxl355_get_raw_temp(adxl355, (uint16_t*)&temp_raw);
+		if (ret)
+			return ret;
+
+		ret = (int32_t)temp_raw;
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &ret);
+
+	default:
+		return -EINVAL;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Handles the read request for converted attribute.
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl355_iio_read_converted(void *dev, char *buf, uint32_t len,
```

Comment:
I'm not sure here, but is this redundant? Do IIO clients calculate this themselves using the raw, offset and scale values?